src/core/reducer.ts

```diff
@@ -294,12 +385,15 @@ export function CreateGameReducer({
 
         if (game.disableUndo) {
           error('Undo is not enabled');
-          return state;
+          return WithError(state, ActionErrorType.ActionDisabled);
         }
 
         const { _undo, _redo } = state;
 
         if (_undo.length < 2) {
+          // TODO(#723): Add an error case here.
+          //  error(`No moves to undo`);
+          //  return WithError(state, ActionErrorType.ActionDisabled);
```

Comment:
Do you need a hand deciding on names for the various commented out errors here and below or do you want to have a go at creating error types for these? Or are you reserving this for a separate PR?

Reply:
I'm fine with either:
1. Taking some help to decide on error types and updating the tests as part of this change.
2. Just leaving this as a TODO to the next PR that starts adding semantics.

Reply:
I think it would be nice to include these as part of this PR so we have the reducer code basically done. I’ll take a look and make some suggestions.

Reply:
OK, so the cases to cover are:

- No moves to undo
- Cannot undo other players' moves
- Move cannot be undone
- No moves to redo
- Cannot redo other players' moves

Do you think collapsing these all into a single error type would be OK?

- `action/action_invalid` — The requested action is not currently possible

It would make handling each specific case separately, but keeping the total number of error types down also seems desirable.

Reply:
So when trying to update the master tests after raising these errors, I found several tests that don't do what they claim to be doing and, in general, that file looks like it needs to be reworked. There's a ton of dependence between different test cases AND different test suites, and they are extremely order dependent.

It's also super unclear what, if any, coverage the seemingly duplicated tests are adding compared to a parameterized test on the game.deltaState field.

Thus, I'm going to do the minimal thing to get these tests passing and covering the updated behavior.

Reply:
Should I raise a bug for ^ and fix that first?

Reply:
Good catch! A fix PR would be ideal (no need to open an issue for something like this).

Reply:
Opened https://github.com/boardgameio/boardgame.io/pull/942 to fix this underlying issue.